app/model/service.py

```diff
@@ -131,14 +131,28 @@ def TryCheckStatus(self):
 		status = 0
 		resultMessage = ''
 		try:
-			if self.data[Service.STATUS_CHECK_STRING]['type'] == 'HTTP':
+			status_check_type = self.data[Service.STATUS_CHECK_STRING].get('type', None)
+			if status_check_type == 'HTTP':
 				r = self.Request()
 				if r != None:
 					status = r.status_code
 					if status != 200:
 						resultMessage = r.text
+			elif status_check_type == 'Heartbeat':
+				required_count = self.data[Service.STATUS_CHECK_STRING]['count'] #how many events...
+				required_type = self.data[Service.STATUS_CHECK_STRING].get('event_type', 'Heartbeat') # of this type
+				tag = self.data[Service.STATUS_CHECK_STRING].get('tag', None) # and possibly with the given tag in the message
+				period = int(self.data[Service.STATUS_CHECK_STRING]['minutes']) #in how many minutes do we expect
+				query = {'type': required_type, 
+						'time': {'$gt': datetime.datetime.now() - datetime.timedelta(minutes=period)}}
+				if tag:
+					query['tags'] = tag
+				count = Event.Count(self.id, query)
```

Comment:
Yes without the right index(es) on the events could be very slow if there are many, many events.
An alternative would be to have a little cache (a queue) for "heartbeat" events that has e.g. the last 10 or so of them, but then that cache should be shared all Monni processes. A cache that is shared between processes is a db by another name, so.... Perhaps a good solution is to have a dedicated, and capped, collection for heartbeats, so that the query in  the status check will be fast, without requiring that the index needs to be large.